### PR TITLE
Auth validations

### PR DESCRIPTION
**Pull request by tanishk15sharma**
**Fixes** #7 

- Refactored Authentication code
- Validations Functions
- Regexp.File
- Login & SignUp input validations messages

### CODE REVIEW

**Review by Mohitkumar33 — APPROVED** [review]

File: src/components/login/Login.css [hunk 1]
```diff
@@ -63,3 +63,9 @@ html {
   margin: 0;
   height: 100vh;
 }
+.err-msg {
+  align-self: flex-start;
+}
+.fa-circle-exclamation {
+  color: #ef4444;
```
Comment (AmishMishra11) [inline src/components/login/Login.css:70]: you could use variables here

File: src/components/login/Login.js [hunk 2]
```diff
@@ -3,65 +3,92 @@ import { useState } from "react";
...
+      authDispatch({ type: "USER_LOAD" });
+      const { data } = await axios.post("/api/auth/login", { email, password });
+      authDispatch({ type: "USER_LOAD_SUCCESS", payload: data.foundUser });
       console.log(data);
       localStorage.setItem("token", data.encodedToken);
     } catch (err) {
       console.log(err);
     }
   };
   console.log(authState);
```
Comment (AmishMishra11) [inline src/components/login/Login.js:30]: remove console before pushing

File: src/components/login/Login.js [hunk 3]
```diff
@@ -3,65 +3,92 @@ import { useState } from "react";
...
-    authDispatch({ type: "USER_LOAD" });
+  const postLoginDetails = async (email, password) => {
     try {
-      console.log(email, password);
-      let { data } = await axios.post("/api/auth/login", { email, password });
-      authDispatch({ type: "USER_LOAD_SUCCESS", payload: data });
+      authDispatch({ type: "USER_LOAD" });
+      const { data } = await axios.post("/api/auth/login", { email, password });
+      authDispatch({ type: "USER_LOAD_SUCCESS", payload: data.foundUser });
       console.log(data);
```
Comment (prathmeshjagtap) [inline src/components/login/Login.js:24]: remove console.log

File: src/components/login/Login.js [hunk 4]
```diff
@@ -3,65 +3,92 @@ import { useState } from "react";
...
+    <form className="bg" onSubmit={handleSubmit}>
+      <div className="login-container center">
+        <h1 className="login-title">Login</h1>
 
         <input
           type="text"
           placeholder="Email"
           name="email"
-          value={userData.email}
+          value={loginData.email}
```
Comment (AmishMishra11) [inline src/components/login/Login.js:50]: you can destructure login data and extract out email and password

File: src/components/login/Login.js (hunk 2 above)
Comment (prathmeshjagtap) [inline src/components/login/Login.js:30]: you can remove the console.log

File: src/components/signup/SignUp.js [hunk 5]
```diff
@@ -2,29 +2,56 @@ import "./SignUp.css";
...
+    setSignUpErrors((errData) => ({
+      ...errData,
+      [e.target.name]: "",
+    }));
   };
-  const postSignUpDetails = async (e) => {
-    e.preventDefault();
-    authDispatch({ type: "USER_LOAD" });
 
+  console.log(signUpData);
```
Comment (AmishMishra11) [inline src/components/signup/SignUp.js:44]: remove console log before pushing

File: src/components/signup/SignUp.js [hunk 6]
```diff
@@ -2,29 +2,56 @@ import "./SignUp.css";
...
+      e.preventDefault();
+      const { isValid, errors } = validSignUp(signUpData, signUpErrors);
+      if (!isValid) {
+        setSignUpErrors(errors);
+        return;
+      }
+      authDispatch({ type: "USER_LOAD" });
+      const { data } = await axios.post("/api/auth/signup", signUpData);
       authDispatch({ type: "USER_LOAD_SUCCESS", payload: data.createdUser });
       console.log(data);
```
Comment (AmishMishra11) [inline src/components/signup/SignUp.js:56]: From here to remove console

File: src/components/signup/SignUp.js [hunk 7]
```diff
@@ -2,29 +2,56 @@ import "./SignUp.css";
...
 
+  console.log(signUpData);
+  const postSignUpDetails = async (e) => {
     try {
-      const { data } = await axios.post("/api/auth/signup", userInfo);
+      e.preventDefault();
+      const { isValid, errors } = validSignUp(signUpData, signUpErrors);
+      if (!isValid) {
+        setSignUpErrors(errors);
+        return;
```
Comment (AmishMishra11) [inline src/components/signup/SignUp.js:51]: What's the use of return here?

**Review by prathmeshjagtap — APPROVED** [review]
Code is well written 
I liked that you are taking care of Errors,
and using Regex for checking emails and passwords.

**Review by AmishMishra11 — APPROVED** [review]
Liked the way how you have provided checks for valid user inputs in your login and signup page.
Made some comments so fix those before pushing.

File: src/components/login/Login.js (hunk 3 above)
Comment (tanishk15sharma) [reply src/components/login/Login.js:24]: This is for testing , I will remove this .

File: src/components/login/Login.css (hunk 1 above)
Comment (tanishk15sharma) [reply src/components/login/Login.css:70]: okkk

File: src/components/login/Login.js (hunk 4 above)
Comment (tanishk15sharma) [reply src/components/login/Login.js:50]: ok

File: src/components/signup/SignUp.js (hunk 7 above)
Comment (tanishk15sharma) [reply src/components/signup/SignUp.js:51]: if This conditions is true I will just return then and there , code will not run further .

**Pull request merged**